include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -155,6 +155,28 @@ __pattern_transform_scan_base_impl(_ExecutionPolicy&& __exec, _Iterator1 __first
         .wait();
 }
 
+template <typename Iterator>
+bool
+__is_equal_iterators_impl(Iterator it1, Iterator it2)
+{
+    return it1 == it2;
+}
+
+template <typename Iterator1, typename Iterator2>
+bool
+__is_equal_iterators_impl(Iterator1 it1, Iterator2 it2)
```

Comment:
I think it is a redefinition of `__is_equal_iterators_impl` in case of Iterator1 and Iterator2 are the same type..
Please use SFINAE approach, like this
https://godbolt.org/z/rGfz7j8nh

Reply:
Fixed